### DISCUSSION TITLE
Interpolation or not of T & S

### DISCUSSION THREAD

**Discussion by mankoff** [NASA-GISS/freshwater-forcing-workshop#17] [Model implementation]
This isn't a question about freshwater, but about how we create freshwater when coupling GCM oceans to ice sheet models.

In our model which does not resolve fjords, this step involves taking the nearest ocean grid cell and extracting the T and S properties that PISM uses as boundary conditions. If this TS comes from a single cell, it's fairly simple. But it could come from `nearest neighborhood` instead of `nearest neighbor`, or it could involved an intermediary bucket that mixes attempts to mix the freshwater and far-field water.

I have two questions, one specific (is interpolating bad) and one broad (how others handle this).

+ Is interpolating bad? I think it should either be avoided, or it gets complicated. Simple interpolation would treat T and S as independent and linearly interpolate between neighbors (perhaps weighted by distance) or between far-field and coastal water masses. However, density is a non-linear function of T&S and therefore interpolating those two independently could create water masses that don't actually exist, or are unstable and require overturning and therefore induce mixing, etc. My hunch is therefore to avoid interpolation, or if it is needed, figure out how to do it not as two independent linear equations, but as `f(T,S,\rho)`.

+ How do others handle this in their ocean/ice sheet coupling?

**Comment by xylar** [comment]
> Is interpolating bad?

For purposes of calculating the thermal forcing, which is presumably what you need, the freezing temperature is a much stronger function of pressure (i.e. depth) than of salinity and thus the thermal forcing (the ambient temperature minus the freezing point) is a pretty weak function of salinity.  I wouldn't worry too much about interpolating T and S separately.  That's what we plan to do.  But you could interpolate thermal forcing at a fixed depth (or pressure) and that might be slightly safer.

Does your ice sheet-ocean coupling scheme actually make use of the ocean density and if so how?  Is it in a way where small differences in density will matter?  I wouldn't expect so.  We use a fixed reference density for ocean water in things like flotation.  But even if you use the "real" ocean density, it won't be appreciably different for purposes of flotation whether you interpolate or just use a nearest neighbor.
